src/lobpcg/eig.rs

```diff
@@ -30,32 +33,36 @@ use std::iter::Sum;
 ///
 /// let res = eig.decompose(3);
 /// ```
-pub struct TruncatedEig<A: NdFloat> {
+pub struct TruncatedEig<A: NdFloat, R: Rng> {
     order: Order,
     problem: Array2<A>,
     pub constraints: Option<Array2<A>>,
     preconditioner: Option<Array2<A>>,
     precision: f32,
     maxiter: usize,
+    rng: R
 }
 
-impl<A: Float + NdFloat + PartialOrd + Default + Sum> TruncatedEig<A> {
+impl<A: Float + NdFloat + PartialOrd + Default + Sum> TruncatedEig<A, Xoshiro256Plus> {
     /// Create a new truncated eigenproblem solver
     ///
     /// # Properties
     /// * `problem`: problem matrix
     /// * `order`: ordering of the eigenvalues with [TruncatedOrder](crate::TruncatedOrder)
-    pub fn new(problem: Array2<A>, order: Order) -> TruncatedEig<A> {
+    pub fn new(problem: Array2<A>, order: Order) -> TruncatedEig<A, Xoshiro256Plus> {
```

Comment:
Instead of returning with `Xoshiro`, the constructor should accept a generic `Rng` as a parameter. Otherwise you won't be able to construct this with any RNG other than `Xoshiro`. Doing this also eliminates the dependency on `rand_xoshiro`

Reply:
I added a `new_with_rng` to specify a different random number generator

Reply:
Somewhat unrelated, but should we really have "default" constructors that use deterministically seeded RNGs? Those are useful for testing but generally in the wild you want to use something derived from `ThreadRng`. This concern also applies to `linfa`

Reply:
The issue isn't with the RNG quality, it's the fact that calling `new` multiple times results in the same set of random numbers being generated. Is that good or bad? I'm not too sure.

Reply:
We could seed the Xoshiro with thread rng but I actually like the idea of having the same guess in each round, it makes things more easy to track. Perhaps as a tradeoff I'll make the `Xoshiro` dependency optional and rename `new` to `new_with_seed` taking an additional seed parameter

Reply:
I don't really want to add another optional dependency/feature, since that adds more things the user needs to worry about. Either make `Xoshiro` non-optional when using `iterative` feature or don't include it at all and only provide `new_with_rng`. If you choose the first option, you'll need to update the docs of `new_with_seed`.